packages/react-native-reanimated/Common/cpp/worklets/WorkletRuntime/WorkletRuntime.cpp

```diff
@@ -51,6 +51,11 @@ static std::shared_ptr<jsi::Runtime> makeRuntime(
   }
 }
 
+
+std::string getJsValueUnpackerString() {
+  return "(function valueUnpacker_Exodus_valueUnpackerJs1(objectToUnpack,category,remoteFunctionName){let workletsCache=global.__workletsCache;let handleCache=global.__handleCache;if(workletsCache===undefined){workletsCache=global.__workletsCache=new Map();handleCache=global.__handleCache=new WeakMap();}const workletHash=objectToUnpack.__workletHash;if(workletHash!==undefined){let workletFun=workletsCache.get(workletHash);if(workletFun===undefined){const initData=objectToUnpack.__initData;if(global.evalWithSourceMap){workletFun=global.evalWithSourceMap('('+initData.__reanimated_workletCodeWrapper+'\\n)',initData.location,initData.sourceMap);}else if(global.evalWithSourceUrl){workletFun=global.evalWithSourceUrl('('+initData.__reanimated_workletCodeWrapper+'\\n)',\"worklet_\"+workletHash);}else{workletFun=eval('('+initData.__reanimated_workletCodeWrapper+'\\n)');}workletsCache.set(workletHash,workletFun);}const functionInstance=workletFun.bind(objectToUnpack);objectToUnpack._recur=functionInstance;return functionInstance;}else if(objectToUnpack.__init!==undefined){let value=handleCache.get(objectToUnpack);if(value===undefined){value=objectToUnpack.__init();handleCache.set(objectToUnpack,value);}return value;}else if(category==='RemoteFunction'){const fun=function(){const label=remoteFunctionName?\"function \"+remoteFunctionName:'anonymous function';throw new Error(\"[Reanimated] Tried to synchronously call a non-worklet \"+label+\" on the UI thread.\\nSee https://docs.swmansion.com/react-native-reanimated/docs/guides/troubleshooting#tried-to-synchronously-call-a-non-worklet-function-on-the-ui-thread for more details.\");};fun.__remoteFunction=objectToUnpack;return fun;}else{throw new Error(\"[Reanimated] Data type in category \"+category+\" not recognized by value unpacker: \"+_toString(objectToUnpack)+\".\");}}\n)";
```

Comment:
this code is crucial for unpacking values to avoid passing the code directly to native counterpart  and potentially getting polluted by attacker is now migrated directly to CPP

Reply:
What is the difference between this code and what is passed [here](https://github.com/ExodusMovement/react-native-reanimated/pull/15/files#diff-67f6b7bb6e539e3d5c9c4e5e8bffe0ec680274829d329e126c5131149a7b784bL60)?

Reply:
I see it's from below. Any way we could have this unminified here?
```suggestion
static const std::string VALUE_UNPACKER_JS_CODE = ""
"'use strict';"
...
"  return (valueUnpacker as ValueUnpacker).__initData.code;"
"}";

```

Reply:
This code is __initData.code itself tho. Or are you asking why not make a constant?

Reply:
Do we really need unminified?

Reply:
No, but it would be easier to review that it's identical code.
Do you have a script so we can verify?

Reply:
you can print the code with `getValueUnpackerCode()` from `valueUnpacker`

Reply:
How does this even work in our production version of Hermes, isn't `eval` disabled by default there?

Reply:
We need to migrate to custom host function for production. But it isn't not this pr scope